### ISSUE TITLE
High Concept / Design brief

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1147]
In #1132 I asked an off topic question, and the resulting answers need to be addressed. If I understand correctly nothing is written down on what this project is trying to accomplish, so the risk is that nothing is going to be accomplished. We need a High Concept or a Design Brief or something along those lines.

To respond to the comments on the other issue

---

---
###### Danimal696 commented:

```
those are pretty mudddy waters what you are asking for, i can see different answers, but firstly have into account this game is a work in progress.
What to expect would be basically the same as DK:

-     Dungeon building
-     RTS and creature simulator
-     Good humour (not yet in)
-     Some fun campaign
```

'Basically the same thing as DK' is a first step. However, there are 2 DK games which are pretty different with some things that both games do right, some things both games do wrong, and many things that are just being done right in one of those games. The result is that if you ask 4 DK fans what they like about Dungeon Keeper, you'll get four different answers. If the OpenDungeons team has all want the same thing from this game that's great, if you want different things you'd best find out early.

---
###### Danimal696 commented:

```
No, we dont have that kind of budget, but we do have an overall design, played&tested
mechanics and most importly, the will to finish this as best as some hobbist ragtag team can.
Thats the reason we love getting feedback, it help us improve; but we wont deviate much from
our main roadmap to cater to everyone tastes. Thats is a certified project killer.
```

It's good to know that on this project designers get paid but developers, modelers, testers, artists and other contributors work for free. :smile:
Seriously though, @Danimal696 please share the roadmap as that's already something.
Where I strongly disagree is your implication that having a design - a clearly defined goal and direction for the project - will make it more likely for you to get sidetracked away from ever finishing. Or that outside requests are your biggest risk.
In fact the opposite is true, if you can clearly articulate what you want, have a High Concept thought out and written down, you can look at each feature that you think of or is ever proposed and ask 'does this feature reinforce the main concept of my game or detract from it'. It will give you focus and it will help you to cut features you thought you wanted but really don't need. And knowing what to cut is the difference between a successful project and a failed one.

---
###### akien-mga commented:

```
There are are no strong design plans at the moment no, I don't know if there were when the
original developers started OpenDungeons, or if they just wanted to implement a DK-like
engine. When we took over the development, we went from an alpha graphics engine to
something that actually looks like a game - @Danimal696 was the guy with the clearest 
view of the design we should aim for I think, but it mostly evolved little by little as we
added what we wanted to add and discussed how it should impact the gameplay.

So of course, it's pretty clear when playing the game that it was development to match a
well structured design; we mostly build it so that we can have some fun playing it, and
hope to get player feedback to see what direction to go on. So far we plan milestones
which are mostly based on necessary features (rooms, creature behaviours, spells, AI,
networking, scripting, campaigns, etc.).

Now, I also think that we need a clearer view of what we want the "final" game to be like, 
and to have answers to the basic questions you ask such as "what is going to engage 
the player" and "where does the fun come from". If you want to help with that, it would 
be very welcome :)
```

@akien-mga I'm willing to help, but you should realize that the people doing all the work are fundamental in what the design is going to become. _I can't tell you what the final product will be like, I can only ask the right questions._ If I'd design the game I'd like to play and you'd all hate nobody would be happy and it won't get done. You're putting in the work, you're not doing this to make money, so the end result should be a game you love to play. And I don't know what that is.
Again, if there's anything specific for me to read up on, show me. (For example how did Danimal696 share his view on the design with the rest of you?)

**What I'll do to help is the following:**
- I'll ask some questions here on this 'issue' that might lead to a High Concept of this game, which would be great to get some answers on.
- I'll be available for feedback, answer questions and will provide input on the current/future mechanics on how they'll impact the overall game.
- I'll occasionally open some other 'issues' with provoking questions on the designs/mechanics which will help you get a clear view on where you are going and where you'll want to go.

---

---
## Questions on the direction of the project.

As I mentioned above I need to know what you all imply when you're saying you want a game 'like dungeon keeper'. A recent example on why this is important is the reception of War for the Overworld, which aspired to be an unofficial dungeon keeper three. The person in charge of that project loved competitive DK2 so he had his team build all kind of features around concepts like 'competitive build orders' and skill based play, but because the communication around this project was basically 'just _like_ dungeon keeper' the people who always player a 'dungeon sim' were quite disappointed by the fast pace of the game. As a small team you need to figure out your own balance, focus on accomplishing this, and communicate these goals internally and externally.
##### My questions for the core team members are:
1. What is it you like about dungeon keeper that you can't find in other games?
2. If somebody doesn't care about the setting of being an evil overlord, why should he still play DK?
3. Do you play dungeon keeper to relax or for adrenaline?
4. If you play a level, do you expect to win by careful planning or by skillful improvisation?
5. Do the heroes exist to test how well you've made a dungeon, or do you make a dungeon to beat the heroes?
6. How long do you want an average play session to last?
7. What is the biggest problem of DK1? What of DK2? (Disregarding technical issues, bugs and graphics).
8. What is more fun in DK2 than in DK1? What's more fun in DK1 over DK2?
9. What aspect of DK do you want to improve upon in OpenDungeons?
10. If you could only have online multiplayer or only a single player campaign/scripted maps in OpenDungeons, what would you choose?
11. What should make it engaging to build your dungeon?
12. What should make it engaging to beat your rival(specifically the combat)?

**Comment by hwoarangmy** [comment]
1 - I would say the humor. That's what we really lack IMHO in this game because it would require some funny animations and that would mean more work on the artists shoulders.
2 - ATM, I think we have reached graphics level that is pretty good for an amateur project (none of us is working as a professional game developper/artist). The game is stable enough to be playable and we have a strong focus on multiplayer game. So, I would say that he can play a funny game with his friends ^^
3 - To relax. However, we had decided that we would aim for games lasting more or less 30 minutes in standard multiplayer games (so we should have most core features unlocked by that time).
4 - Depends on the level. I would say that we will probably want to have skirmish levels that will require each type of skills to be won (spells/traps/creatures). Concerning multiplayer levels, we aim at a strategic game so there might be several ways to win. IMHO, DK was too easy. For my part, I used to go straight for a vampire (with warlocks in the library then slapping them to death to get a vampire). Then, I only had to level up the vampire. One of them was enough to win nearly all levels. That's something we should avoid in OD. For this reason, we divided the creatures 3 per tier (there are 4 tiers). We have balanced their skills so that one of each tier is specialized in one attack type (physical, magical, elemental). Basically, we created a triangle and each creature can win versus a creature of the same tier. That means that if you adapt your creatures to your opponent, we get an advantage.
5 - I don't remember the heroes concept in DK but in OD, heroes come to your dungeon to steal your gold. Fighting the evil is a nice side effect but they are here for gold first ^^
6 - 30 minutes average in a standard size level. IMHO, more will be too slow for multiplayer games.
7 - DK1: too easy and too repetitive. As I said, I could win every level by going straight to vampires and turtling until the vampire was ready to crush everything nearby.
DK2: maybe lacking multiplayer game.
8 - Graphics were better in DK2. For some reason, I think DK1 humor was better than DK2 (I didn't play any of those games for years, though).
9 - Whatever we can ^^ But IMHO, the strategic aspect is the most important to improve. In the original concept of OD, there were some paths (basically, spells/creatures/traps). ATM, we have not really polished how the paths would really work.
10 - As an open source project, I think multiplayer is what we need to focus on. However, campaigns are good to help new players to learn how to play. And, ideally, if we can find a nice story board, that would be cool.
11 - To get better creatures/traps/spells
12 - To steal his ressources ^^

Note that there is a global creature issue that might answer some of your questions #1081 

I understand your will of trying to find clear goals but here, it is a spare time project. The issues we open here are to be picked up by volunteers. We define the milestones we aim for but we change at every version depending on what has been done. Usually, when one of us has an idea, he just opens an issue to discuss it until we reach a consensus. I believe that if one of us was to try to be the leader and to rule everything, the other team members would leave. ATM, I have fun working on the game and that's why I do it. If I was told what to do and how to do it, I'm pretty sure I wouldn't stay and I guess it is the same for the others. IMHO, that's the biggest difference between professional games (where you can say everybody what to do since you pay them for that) and a spare time project (where you need to let people do what they like or they might go away).

**Comment by Danimal696** [comment]
When the last team left, the project was a real mess, most of it was just some proof of concept mechanics and lots of unrealistic goals. At first i simplified and shaped it a lot to resemble more the original DKs, after that i stepped down and everything is decided among all, my opinion may have some weigth but its not final or decisive, so i dont consider myself the project leader, just the older member around. 
A formal grand roadmap is hard to provide since everything is spread out in the forum, which is why any newcomer might get lost, also every "department" works on what they want or percieve most pressing at the moment. The code guys keep on adding new functionalities and improving the existants ones, the art guys (me and Eugeneloza) have tasked ourselves with improving all graphics, creatures renewal goes first, then any model to complete the last piece of added code, after that, there is no shortage of things to improve graphically... and other times its the other way around  and they code to fit something new i did.
Overall, we dont like to stress ourselves with hard timelines or strict jerarquy, i think we have enough of that in our life (i have like 5 minor bosses and also a few major ones at work); but you can see  the current short term goals here:
https://github.com/OpenDungeons/OpenDungeons/issues/849
- What is it you like about dungeon keeper that you can't find in other games?
  The dark humour and lacks of morals, i loved torturing the shit of everything i captured and raising and army of the dead.
- If somebody doesn't care about the setting of being an evil overlord, why should he still play DK?
  Because its a nice and engaging strategy game.
-  Do you play dungeon keeper to relax or for adrenaline?
  Relax, building and investigating in single payer, to bash faces in multiplayer
- If you play a level, do you expect to win by careful planning or by skillful improvisation?
  Careful planning, thats what DK is about, you dont have a good dungeon and hence good creatures if you dont plan well.
- Do the heroes exist to test how well you've made a dungeon, or do you make a dungeon to beat the heroes?
  Both, your dungeon has to both hold whatever is thrown at it and be to able to dish pain around.
- How long do you want an average play session to last?
  Short on multiplayer, at consumer taste on single (raising army of dead!)
- What is the biggest problem of DK1? What of DK2? (Disregarding technical issues, bugs and graphics).
  Overpowered creatures and sure to win tactics.
- What is more fun in DK2 than in DK1? What's more fun in DK1 over DK2?
  I prefered the more grim DK1 with easy to anger and very expendable creatures,in Dk2 the nice graphics but it was easier and slower paced,
- What aspect of DK do you want to improve upon in OpenDungeons?
  Giving more different playing styles choices
- If you could only have online multiplayer or only a single player campaign/scripted maps in OpenDungeons, what would you choose?
  Both go together, being the single mode the hardest to acomplish.
- What should make it engaging to build your dungeon?
  The challenge of keeping it together
- What should make it engaging to beat your rival(specifically the combat)?
  To crush your enemies. See them driven before you. Hear the lamentations of their women. 
  - I believe that if one of us was to try to be the leader and to rule everything, the other team members would leave. ATM, I have fun working on the game and that's why I do it. If I was told what to do and how to do it I'm pretty sure I wouldn't stay

Exactly my thoughs, i dont want to rule or be ruled, this is for fun even if we are serious about it

**Comment by Loobinex** [comment]
###### hwoarangmy commented

```
1 - I would say the humor. That's what we really lack IMHO in this game because it would require some funny animations and that would mean more work on the artists shoulders.
2 - ATM, I think we have reached graphics level that is pretty good for an amateur project (none of us is working as a professional game developper/artist). The game is stable enough to be playable and we have a strong focus on multiplayer game. So, I would say that he can play a funny game with his friends ^^
7 - DK1: too easy and too repetitive. As I said, I could win every level by going straight to vampires and turtling until the vampire was ready to crush everything nearby.
DK2: maybe lacking multiplayer game.
10 - As an open source project, I think multiplayer is what we need to focus on. However, campaigns are good to help new players to learn how to play. And, ideally, if we can find a nice story board, that would be cool.

I believe that if one of us was to try to be the leader and to rule everything, the other team members would leave. ATM, I have fun working on the game and that's why I do it. If I was told what to do and how to do it, I'm pretty sure I wouldn't stay and I guess it is the same for the others.
```

@hwoarangmy So (1) a funny (10) multiplayer focused game. That would be quite a challenge I think as comedy is much easier to do in a single player/story setting than it is in a multiplayer game. When you have a 30-minute battle with a friend, you're not exactly wanting to watch your creatures dance when the casino has a jackpot right? So do you have any idea how you'll wanna get there?

And DK1 being too easy is a symptom of underlying problems, not a problem in itself. The main game was very light on content, just a single 20 level campaign which spent a lot of time teaching you the game. The later levels where challenging enough for newcomers, but only the first time you played it. The Deeper Dungeons expansion showed some of the real problems, where it showed that some of the design problems made it very difficult for level designers to show some proper levels. (For example prison+torture room are the most fun rooms in the game, but they are so effective in adding the strength of the enemy to your own that giving the enemy more creatures doesn't provide more of a challenge.) It is possible though, try [this map](https://keeperklan.com/downloads.php?do=file&id=113) and you'll probably lose.

Having a design does not mean having a leader. A good developer is worth his weight in gold and designers and testers are there to support him, not to lead him. Yes, when you're told to do something you don't like to do on your hobby you'd be silly to do so, that's why I said the design should be their to support _your_ vision of the game. But also, when you spend a lot of time on the game and it is shaping up to be something that just doesn't work, is no fun to play, you'll also quickly lose motivation. I've yet to meet a developer who enjoys it when he finds out that what he has built can be thrown away because it is not what the customer wanted/is no fun/doesn't fit with the rest of the product.

###### Danimal696 commented

```
Overall, we dont like to stress ourselves with hard timelines or strict jerarquy, i think we have enough of that in our life (i have like 5 minor bosses and also a few major ones at work); but you can see the current short term goals here: #849
```

@Danimal696 I'm not saying you need more leadership, and I'm certainly not saying I'm should be the boss. I've been the major boss and the minor boss, and I'm retired now so I'm also not looking for that. What I'm saying is that it would be very nice for everybody involved if all features, mechanics and other work that is done serves some overall design and benefits the end result. That everything fits together well.

```
What should make it engaging to beat your rival(specifically the combat)?
    To crush your enemies. See them driven before you. Hear the lamentations of their women.
```

That doesn't help, yes winning is nice but you just said having the game be very easy is no fun. So let me rephrase, what should make the combat fun? How are you as a player involved in the victory? Or do you want a single button that says 'attack' and do you sit back and wait to see if you win or lose?

**Comment by hwoarangmy** [comment]
> So (1) a funny (10) multiplayer focused game. That would be quite a challenge I think as comedy is much easier to do in a single player/story setting than it is in a multiplayer game. When you have a 30-minute battle with a friend, you're not exactly wanting to watch your creatures dance when the casino has a jackpot right? So do you have any idea how you'll wanna get there?

Yes, of course, looking at your creatures will be much more on purpose during the campaign than during the game. I would also had that once you have seen the black mistress in the torture room once, you won't find it as funny the next time. But you will remind that it was fun. And that's what I think we should go for.

> And DK1 being too easy is a symptom of underlying problems, not a problem in itself

Well, difficulty depends on the map. Changing them is not too hard. What is difficult is to balance the game so that it is chalenging in multiplayer. Making sure that there is no unique strategy that beats all the others and that every player will go for. That's what is hard. For this reason, we have set in config files (text mode) the description of most features. This way, if someone wants to try to balance the game, he doesn't have to be a developper and to compile the game. Changing the config files is enough.

**Comment by Danimal696** [comment]
> @Danimal696 I'm not saying you need more leadership, and I'm certainly not saying I'm should be the boss.

well, i dont consider myself a boss at all (and dont want to be) and your feedback is pretty welcome, im not taking any of it as an offense. Anyways, putting something like that together will take a lot of time and effort but may be worth it, currently the wiki tries to fill that spot.

>    To crush your enemies. See them driven before you. Hear the lamentations of their women.

Thats a Conan quote! how cant you not know it? heresy...
To me the answer to that is adapting my dungeon and creatures, the enemy has lots of melee? ill make big libraries and get plentyful of mages, strategies like that are what make the combat engaging and the player feel in command, if you mess with your dungeon design you will regret it. In resume, to feel your decisions are the ones that made you win even if creatures kill each other by themselves.

**Comment by Loobinex** [comment]
@Danimal696 The High Concept does not take that much time at all. It should fit on a single page of text. Naturally speccing out the game further will take a lot of effort and that will go along side the development of the game.

It sounds to me like both of you prefer the dungeon management aspect of the game, and that combat in itself is not important.
How would you feel about having 'neutral zones' on the maps, land where your creatures and rival creatures can go, but you can't drop creatures and can't pick them up? When these are the main contested area's it will put additional emphasis on your creatures being an independent army and you'll point them to where to go. This makes a barracks room prominent where you can make squads out of several creatures that will move and fight together.

**Comment by Danimal696** [comment]
Theres already such a mechanic, when invading enemy dungeons you cant drop creatures nor in unclaimed lands, there is a spell that makes all ready creatures rush there "call to arms".

**Comment by Loobinex** [comment]
Yes I know, just like both dungeon keepers. However, this can all be claimed, and most attacks are begun by claiming first and then fighting on the edge where red and blue land meet. Heroes are almost exclusively battled on claimed ground. This means almost always you'll just drop all your creatures directly on the battle.
I mean a type of land that cannot be claimed ever but can be walked over.

**Comment by Danimal696** [comment]
We have :), rock 0% fullness

**Comment by Bertram25** [comment]
My turn I guess. (Been pretty busy, but well, I like design discussions :])

Anyone, feel free to correct me if I'm wrong, but here is my opinion:

@Loobinex If I may, I think this issue is pretty cool to summarize a bit why every people involved around is still around. :)
But overall, this issue can be summarized in one of your sentence:

> What I'm saying is that it would be very nice for everybody involved if all features, mechanics and other work that is done serves some overall design and benefits the end result. That everything fits together well.

The good thing is that you care enough to try and refine the design goals of this game, the thing I'm trying to understand though is: Basically, you have understood we're working on a game heavily inspired by DK1/2 games, but we don't exactly want to make a clone. So yes, we want to take the essence out of it, and refine it into our own beast.

I will be straight but, everyone here has a growing experience in what he wants about the DK game, and I think we already had some discussions about the 3 builds type of creatures in combat, the different strategies (dungeon building planning, rush strategy possibility, use of strategic point on the map) and so on.
I think we also all agree (more or less) about the fact that the game core feature are there, and that the next steps is balance fine-tuning and the addition of scripting support and campaign support, which are the developers last big pieces. The rest is balancing and careful level design to me.

Having other grand plans is great on paper to me, but on a spare time projects, it may hurt the projects, especially the design changes as we see fit, because implementing the desired features can take (much more) time and because people change their mind on what is fun over time.
So when I restarted the project development with the help of @Danimal696 and when @akien-mga  and @hwoarangmy jumped on board most notably, we tried to go step by step by discussing what is the next small set of features we find fun, without any greatly defined grand plan, since, again, we all agree it's yet another game where "it's so good to be bad"(tm) and where you build dungeons, and push dark critters toward your enemy with the most delighted pleasure.

The game is quite fun already and we know we lack a campaign mode and good well-balanced fun maps. Is there anything else you need to know?

**Comment by Loobinex** [comment]
@Bertram25: Thanks for the reply.
Again though, a 'High Concept' or a 'Brief' is just the opposite to having a 'greatly defined grand plan', the former is at most a page of text and more importantly just articulated the direction you want to go in and the latter sounds like a lot of paperwork to me.
I think Dungeon Keeper is a bit of an unusual game, in the sense that it is one of a kind and almost everybody feels like it is about something else. If you think all of you are on the same page already and know what kind of experience your going for, than that's what counts. As long as you realize that while there's certainly a craving for more 'it's good to be bad' games they were almost universally met with disappointment because the games did not capture what the players were looking for.

I apparently misunderstood the current status of the project, when I played it I assumed you were in the process on getting an engine running and had focused purely on technical issues. I did not know the current build is already considered fun. (I'm sorry, this sounds mean, it's not intended that way.)
It felt to me that a lot of mechanics subvert the what the game should be about, and counter what Danimal696 and hwoarangmy state above. My overall experience is that building a dungeon is trivial, with no thought going into it I got the most powerful creatures and did not run into any problems with my layout.

In any case, I've gotten plenty of information on this issue that I could have otherwise gotten from a design brief. Enough to be able to raise some topics that should give you some food for thought.

**Comment by Bertram25** [comment]
> I apparently misunderstood the current status of the project, when I played it I assumed you were in the process on getting an engine running and had focused purely on technical issues. I did not know the current build is already considered fun. (I'm sorry, this sounds mean, it's not intended that way.)

Lol, well, it's so good to be bad, right? ;P

> It felt to me that a lot of mechanics subvert the what the game should be about, and counter what Danimal696 and hwoarangmy state above. My overall experience is that building a dungeon is trivial, with no thought going into it I got the most powerful creatures and did not run into any problems with my layout.

Now there is something interesting, if you can pinpoint what is "too easy", we can maybe work on that point per point.

> In any case, I've gotten plenty of information on this issue that I could have otherwise gotten from a design brief. Enough to be able to raise some topics that should give you some food for thought. 

Good! I guess the future of this issue belongs to forums or more precise tasks, then?

**Comment by Loobinex** [comment]
When I have some time I indeed want to raise some topics about dungeons and creature concepts.

I'll leave the administration of closing issues and moving discussions to other places to you. However, if you feel you'd like to have High Concept, I have received input for it here so I could provide one for you all. If you don't see any value I won't make one as it is a bit of work.

**Comment by hwoarangmy** [comment]
Note that ATM, we are lacking room diversity. When we will have more models, we might be able to have creatures depending on rooms (like in DK).

Since you are talking about concepts, to increase the strategic aspect of the game, for the first attempt at balancing, I've made tier 4 creatures specialized (physical, magical and elemental) depend on 2 corresponding tier 3 creatures. This way, switching from one specialized type to another is not (or at least less) trivial and makes checking your opponent dungeon a bigger priority.

Regarding this point, when the 0.7.0 is out, I might have a look at implementing a scout spell that shows some target tiles away.

**Comment by Loobinex** [comment]
I have started making a big post about creature attracting which I'll post on the forum when it's done and my account is approved there. That's not what I mean by dungeon building here,...
What I'm talking about is that on the two levels I've played I just made a clean grid of 5x5 rooms and whenever I researched a new room I just build it straight away. It does not seem to matter where to build rooms, how to fit them on the level, which rooms where or what size to build them.

In Dungeon Keeper the first few minutes of each level you can plan your dungeon, to build the optimal layout, and you'll have to consider things like not having creatures walk through libraries to not upset your researchers and have them shoot at your workers, to strategically place your lairs to not instigate fights between creatures that hate each other, have your workshop close to your hatchery and training room because bile demons walk very slowly, build everything in an efficient shape to be able to reinforce the walls quickly for security and room efficiency. Stuff like that.

**Comment by hwoarangmy** [comment]
ok. That is planned and more or less implemented. Infortunately, for balancing purposes, we didn't use it much. ATM, only spiders/cavehornets fight each other when there are too many of them together. This has been done almost only to remind how it should be done.
If you want to give a try, you can have a look at config/creatures.cfg. You can add mood modifiers that will make creatures get angry if they see the given creature type.
Then, one a creature is angry, it will engage natural ennemies. You can have a look how it is done in the config file.

Note that you don't need to compile or build anything to test that.
Creature speed can also be changed there.
